pages/story/[id].tsx

```diff
@@ -25,19 +25,38 @@ import { list, get } from '../../lib/api/stories'
 import StoryDetail from '../../components/stories/StoryDetail'
 import { storyCite } from '../../components/stories/model'
 import FloatingRibbon, { Button } from '../../components/common/FloatingRibbon'
+import { ResponseError } from '../../lib/errors'
+import { GetStaticPropsResult } from 'next'
+import ErrorPage from '../_error'
 
-interface Props {
+interface StoryProps {
   story: Story
-  url: string
+  url?: string
 }
 
+interface ErrorCodeProps {
+  errorCode: number
+  errorMessage: string
+}
+
+type Props = StoryProps | ErrorCodeProps
+
 const shareIconSize = 64
 const contentSize = 150
 
-export default function StoryPage({ story }: Props) {
+export default function StoryPage({ story, errorCode, errorMessage }: Props) {
```

Comment:
I think the type here should be `StoryProp` correct? And the `StoryProp` interface needs to have `errorCode` and `errorMessage` added as well.

Reply:
I separated the two interfaces to prevent someone from passing, for example, both an `errorCode` and a `story`. The problem was that TypeScript couldn't tell which one of the two interfaces was passed, so I have refactored it a bit and added a `success` prop which allows to easily distinguish between the two interfaces, making it a _tagged_ union type.
Let me know what you think!